### PR TITLE
[Docs] New gas doc and a minor nit in sys integ doc.

### PR DESCRIPTION
**Pull request by rajkaramchedu**
@chloeqjz hey, see the nit fixed. Thanks!

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4464)

### CODE REVIEW

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 1]
```diff
@@ -2,71 +2,122 @@
...
+# Gas and Transaction Fees
 
-# Introduction
+## Concept
 
-Gas ensures that all Move programs running on the Aptos blockchain eventually terminate. This bounds the computational resources that are used. Gas also provides the ability to charge a transaction fee, partly based on consumed resources during execution.
+To conduct any transaction with the Aptos blockchain on the mainnet, you are required to pay a processing fees. This includes transactions from a client application, or from a stake owner, a node operator or a voter. Think of this processing fee in the following terms: 
 
-When a client submits a transaction for execution to the Aptos blockchain, it contains a specified:
+Any blockchain, for example, the Aptos blockchain, exists in a distributed network of computing and storage resources. The Aptos mainnet is one such distributed network, holding the Aptos blockchain.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:12]: I think I mentioned this in the doc, the phrasing here is strange.

`Most blockchains including the Aptos blockchain exist in a distributed network comprised of computing and storage resources.  The Aptos mainnet is one of these networks.`

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 2]
```diff
@@ -2,71 +2,122 @@
...
+To conduct any transaction with the Aptos blockchain on the mainnet, you are required to pay a processing fees. This includes transactions from a client application, or from a stake owner, a node operator or a voter. Think of this processing fee in the following terms: 
 
-When a client submits a transaction for execution to the Aptos blockchain, it contains a specified:
+Any blockchain, for example, the Aptos blockchain, exists in a distributed network of computing and storage resources. The Aptos mainnet is one such distributed network, holding the Aptos blockchain. 
 
-* `max_gas_amount`: The maximum amount of gas units that can be used to execute the transaction. This bounds the computational resources that can be consumed by the transaction.
-* `gas_price`: The gas price in the blockchain's utility token. Gas price is a way to translate from gas units (the abstract units of resources consumed by the virtual machine) to a transaction fee in the blockchain's utility token.
+The processing fee you are required to pay is based on the computing and storage resources in the Aptos mainnet to:
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:14]: Maybe, `based on the computing and storage resources you use on the blockchain to:`

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 3]
```diff
@@ -2,71 +2,122 @@
...
 
-The transaction fee charged to the client will be at most `gas_price * max_gas_amount`.
+1. Process your transactions on the blockchain.
+2. Propagate the validated record throughout the distributed network of the mainnet, and
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:17]: I would remove the `, and`

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 4]
```diff
@@ -2,71 +2,122 @@
...
+3. Store the validated record in the distributed blockchain storage.  
 
-The gas price, and hence the transaction fee, should follow the market characteristics of the Aptos blockchain as resource supply and demand fluctuate.
+:::tip 
+Conceptually, this fee can be thought of as quite similar to how we pay for our home electricity or water utility.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:21]: `home electric or water utilities`

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 5]
```diff
@@ -2,71 +2,122 @@
...
-## Types of resource usage
+You can even bump up your transaction to a higher priority level on the blockchain by paying a larger processing fee. In your transaction, you can commit to paying a gas price that is higher than the market gas price. This is one way you can move your transaction higher in the priority list and have it  processed quicker. 
 
-For the virtual machine (VM) to execute a transaction, the gas system needs to track the primary resources used by the transaction. These fall into three resource dimensions:
+### Unit of gas
 
-1. The computational cost of executing the transaction.
-2. The network cost of propagating the transaction through the Aptos ecosystem.
-3. The storage cost of data created and used during transaction execution on the Aptos blockchain.
+You can make a simple transaction, or a complicated transaction that requires the blockchain to perform lots of computation and distributed storage. In either case, you will be required to spend a processing fee sufficient to complete the transaction. This is where the notion of **gas** becomes clear**.** Here is how it works:
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:30]: I appreciate the boldness, but I'm not sure a period needs to be bolded?

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 6]
```diff
@@ -2,71 +2,122 @@
...
+  ```
+  (total gas units consumed) * (gas_price)
+  ```
+  must not exceed `max_gas_amount`, or else the transaction will abort the execution.
 
-## Using gas to prioritize a transaction
+The transaction fee charged to the client will be at the most `gas_price * max_gas_amount`.
 
-When you send a transaction, it is prioritized based on different criteria. One of these is the normalized gas price for the transaction.
+## Gas parameters set by the governance
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:64]: `Gas parameters set by governance`

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 7]
```diff
@@ -2,71 +2,122 @@
...
 
-For example:
+The following gas parameters are set by the Aptos governance.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:66]: It's very strange to say `by the Aptos governance` maybe `by Aptos governance` instead.

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 8]
```diff
@@ -2,71 +2,122 @@
...
+The following gas parameters are set by the Aptos governance. 
 
-* Bob sends a transaction with `gas_price` 10.
-* Alice sends a transaction at the same time with `gas_price` 20.
+:::tip On-chain gas schedule
+These on-chain gas parameters are published on the Aptos blockchain at `0x1::gas_schedule::GasScheduleV2`.
+:::
 
-Alice's transaction would be ranked higher than Bob's.
+- `lower bound` : The `max_gas_amount` value in the transaction must be set to significantly greater than this parameter’s value.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:72]: I don't think `significantly` is necessary, but it does need to be set greater, and I guess explain somehow this is to handle adjustments in dynamic pricing / based on how much you are willing to pay total.


This section reads weird to me, though i am reading in the PR system

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 9]
```diff
@@ -2,71 +2,122 @@
...
+- `upper bound` : The maximum allowed value for the `max_gas_amount` gas parameter.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:73]: These names aren't `upper bound` they are...
`txn.maximum_number_of_gas_units` -> maximum number of gas units that can be spent
`txn.min_transaction_gas_units` -> minimum number of gas units that can be spent

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md [hunk 10]
```diff
@@ -2,71 +2,122 @@
...
-| The resource usage of a program needs to be agreed upon in consensus | This means that the method of accounting for resource consumption needs to be deterministic. This rules out other means of tracking resource usage, such as cycle counters or any type of timing-based methods as they are not guaranteed to be deterministic across nodes. The method for tracking resource usage needs to be abstract. |
+Aptos gas pricing uses dynamic prices for the storage operations. This means that the storage costs, hence the `gas_price`, can increase exponentially as the Aptos blockchain state database is filled up. The storage cost can become as high as `100x` at 100% utilization. However, the expectation is that the validators will make use of larger and cheaper storage hardware to mitigate such exponential rise in the storage costs.
```
Comment (gregnazario) [inline developer-docs-site/docs/concepts/basics-gas-txn-fee.md:77]: This is the `gas_used` not the `gas_price` that increases.

The `gas_price` is determined by the market

The `gas_used` is determined by the resources

File: developer-docs-site/docs/concepts/basics-gas-txn-fee.md (hunk 5 above)
Comment (rajkaramchedu) [reply developer-docs-site/docs/concepts/basics-gas-txn-fee.md:30]: Haha, I like the pun. That wasn't intended.

**Review by gregnazario — APPROVED** [review]

**Pull request merged**